sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -114,21 +114,22 @@ export interface CollectSpecificOptions {
   recipient: string
 }
 
-export interface TransferOptions {
+export interface MigrateSpecificOptions {
   /**
-   * The account sending the NFT.
+   * The additional currency and amount that needs to be transferred if migrating (a) from out-of-range to in-range, or (b) from out-of-range to out-of-range on the opposite side
    */
-  sender: string
+  currencyAmount?: CurrencyAmount
+}
 
+export interface CurrencyAmount {
   /**
-   * The account that should receive the NFT.
+   * The additional currency that needs to be transferred
    */
-  recipient: string
-
+  inputCurrency: Currency
```

Comment:
Can you explain why there is a need for a 3rd currency specification? If we're migrating from v3 we can only ever migrate to a pool that contains the same currencies no?

And if we detect the pool is weth, we can just auto-unwrap to eth?